tests/target/configs/format_cargo_toml/false/Cargo.toml

```
@@ -0,0 +1,48 @@
+# rustfmt-format_cargo_toml: false
+
+  # comment before [[bin]], not necessarily at the beginning
+
+
+    [[bin]]    # comment  after [[bin]]
+        "aa" = 1     # comment for "aa"    
+        'bb' = 2 
+        "啊"=1
+
+         # comment before package  
+
+[package]
+    version = 1
+    description = "a\nb\nhaha"
+    name = 3
+
+   # comment 1  for arr1 line1
+   
+# comment 1 for arr1 line 2
+    arr1 = [1,
+        2,3]  
+
+# comment 2   for arr2   
+
+    arr2 = ["11111111111111111111111111111111111111111111111111111111111111111111111111111111","1111111111111111111111111111111111111111111111111111111111111111111111111111111"]
+
+   # comment before [[dependencies]], not after [package]  
+
+
+[dependencies]
+        extremely_long_crate_name_goes_here = {path = "extremely_long_path_name_goes_right_here",version = "4.5.6"}
+
+        
+        crate1 = {            path = "crate1",version = "1.2.3"                   }
+   
```

What about testing these? I’m curious to see how this would end up formatted:

```suggestion
crate2 = { version = "3.2.1", features = ["single_feature"], optional = true }
crate3 = { version = "3.2.1", features = ["a", "few", "more", "features"], optional = true }
crate4 = { version = "3.2.1", features = ["many", "many", "many", "many", "many", "many", "many", "many", "many", "many", "features"], optional = true }

```

---

Yeah, I'm glad to add the tests if you send a PR.

---

Great! Here it is: https://github.com/xxchan/rustfmt/pull/3